### PR TITLE
SW-2917 make plants dashboard work

### PR DESCRIPTION
**Pull request by karthikbtf**
- use number parser
- use number formatter
- create chart only when all data is available.. to avoid constructing charts in parallel

### CODE REVIEW

File: src/components/Plants/DashboardChart.tsx [hunk 1]
```diff
@@ -23,13 +23,36 @@ export interface DashboardChartProps {
...
+
+interface DashboardChartContentProps {
+  chartId: string;
+  chartLabels: string[];
+  chartValues: number[];
+  minHeight?: string;
+  locale: string;
+}
+
+function DashboardChartContent(props: DashboardChartContentProps): JSX.Element {
```
Comment (karthikbtf) [inline src/components/Plants/DashboardChart.tsx:50]: moved the original chart into a separate component that requires a locale and the values/labels..
the parent component/wrapper now just returns null if they are not set.
this handles most cases of useEffect concurrency issues where chart was being constructed for same canvas multiple times in parallel

File: src/components/Plants/PlantingSiteDetails.tsx [hunk 2]
```diff
@@ -125,12 +130,12 @@ export default function PlantingSiteDetails(props: PlantingSiteDetailsProps): JS
...
           const plantsPerSpecies: { [key: string]: number } = serverResponse.reduce((acc, population) => {
-            totalPlantsOfSite = +totalPlantsOfSite + +population.totalPlants;
+            totalPlantsOfSite = +totalPlantsOfSite + numericParser.parse(population.totalPlants);
             if (acc[population.species_scientificName]) {
               acc[population.species_scientificName] =
-                +acc[population.species_scientificName] + +population.totalPlants;
+                +acc[population.species_scientificName] + numericParser.parse(population.totalPlants);
             } else {
-              acc[population.species_scientificName] = +population.totalPlants;
+              acc[population.species_scientificName] = numericParser.parse(population.totalPlants);
```
Comment (karthikbtf) [inline src/components/Plants/PlantingSiteDetails.tsx:138]: here they were all being NaN if the totalPlants was formatted (large numbers)

**Comment by constanzauanini** [comment]
I tested in vercel and the first time loaded, the second chart desn't have data:
Screenshot 2023-02-08 at 15 54 12

if changing and going back:
Screenshot 2023-02-08 at 15 55 23

**Review by constanzauanini — APPROVED** [review]
Looks good now!

**Comment by karthikbtf** [comment]
> Looks good now!

i'll get this in now... we can cleanup the error again later.. after things settle a bit

**Pull request merged**